agent/taskresource/cgroup/cgroup.go

```diff
@@ -88,9 +94,18 @@ func NewCgroupResource(taskARN string,
 // GetTerminalReason returns an error string to propagate up through to task
 // state change messages
 func (cgroup *CgroupResource) GetTerminalReason() string {
-	// for cgroups we can send up a static string because this is an
-	// implementation detail and unrelated to customer resources
-	return resourceProvisioningError
+	if cgroup.terminalReason == "" {
```

Comment:
Details about cgroup creation failure was not surfaced for the same reason as mentioned in the comment. It is mostly an implementation within agent and customers are unaware of it. I don't believe we need to surface it :)

Reply:
Does that apply to volume errors as well? #1631 mentions volume and cgroup.

Reply:
Volume is still a resource that customer configures through task definition, so it may be useful.

Reply:
okay, I will change it to only volumes